meeting-notes/2022-05-14.md

```diff
@@ -0,0 +1,261 @@
+# 5/14/22
+
+@pcohen, @rntz
+
+## TODOs
+
+- [ ] Phil JSON:
+  - add cursor selection ranges to JSON output
+- [ ] rntz JSON:
+  - Add selections (including the anchor)
+  - Multiple cursors
+  - Buffer filename should be null if nonexistent
+  - camelCase
+
+@rntz sez: "maybe we should put all these files in a ~/.cursorless/ directory?"
+
+## Notes
+
+**what phil's fork of the cursorless vscode plugin does**
+
+1. writes hats to a file when there's a change to those hats
+
+**phil next steps**
+
+- Migrate "watch JB/Emacs editor state and tell VS Code to navigate to that same
+  location (thus updating the hats)"logic to cursorless fork
+  (file watcher)
+
+- Make new JetBrains plugin also publish standardized \~/.editor-state.json,
+  since it\'s current JSON is overly complicated for this use case
+
+- Export forked changes in knausj VSCode command server
+
+- Export changes from custom command server
+
+**rntz next steps**
+
+- ~~Install and get building (./build)~~
+
+- Get emacs reading \~/.vscode-hats.json and applying hats
+
+- Think about how commands will be sent to visual studio code\
+  *issue: *vscode "command server"waits for a keystroke to tell it to read a
+  file talon writes to execute cursorless commands. just use a file watcher?
+
+    - Get the knausj VSCode\* actions (most importantly
+      vscode_with_plugin_and_wait) to work without VS Code being focused (ie
+      from the talon REPL)
+
+*hooks I might need:*
+
+https://www.gnu.org/software/emacs/manual/html_node/elisp/Change-Hooks.html
+
+- buffer contents change: after-change-functions
+
+- view changes
+
+- current buffer changes
+
+- cursor/selection change
+
+post-command-hook?
+
+**both next steps**
+
+- Think about temporary files for contents
+
+**things to standardize:**
+
+- paths of JSON files
+
+    - \~/.editor-state.json
+
+    - \~/.vscode-hats.json
+
+- schema of JSON files
+
+**Old (before 5/14/22)**
+
+- serialize editor state to file, at the moment:
+
+- a serial number
+
+- current file path
+
+- top & bottom visible lines
+
+edge case: long lines without wrapping, does cursorless hat them?
```

Comment:
I believe it does; we should probably stop hatting them past a certain length for performance, but I don't believe we do so today